### DISCUSSION TITLE
A mode-aware colors plugin to style both light and dark modes with a single class

### DISCUSSION THREAD

**Discussion by JavierM42** [tailwindlabs/tailwindcss#9874] [Show and tell]
I always found the `dark:` modifier a bit clunky when I have a separate light and dark color palette.

With the plugin, `bg-primary` can be used instead of `bg-primary-light dark:bg-primary-dark`.

https://github.com/JavierM42/tailwind-mode-aware-colors/

Any ideas are welcome!

**Comment by xementor** [comment]
so is there any performance drawback?

**Reply by JavierM42** [reply]
On the generated CSS, I can't think of any drawback. Base Tailwind uses CSS variables already.

As far as the build process goes, it obviously needs to run the plugin so that's a non-zero cost, but it's fairly fast and grows linearly with the amount of colors in the theme.

**Comment by wolfspyre** [comment]
I really like this concept. I'm pretty frustrated by the level of repetitious verbosity required to specify light/dark variants everywhere... having a more elegant way of expressing this is a welcome change
